src/components/tooltip/Tooltip.vue

```diff
@@ -331,7 +331,9 @@ export default {
         if (this.appendToBody) {
             removeElement(this.$data._bodyEl)
         }
-        this.controller.abort()
+        if (this.controller != null) {
```

Comment:
we need:
`this.controller != undefined || this.controller != null`

If `window` is not accessible - `controller` would be undefined, too - not null

Reply:
Thanks for your review. But we do not have to be as strict as you suggested:
1. `this.controller != null` turns `false` if `this.controller` is either `null` or `undefined`. Please see [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null#difference_between_null_and_undefined).
2. `this.controller` is safely evaluated as `undefined` even if `this.controller` has not been assigned. [`typeof` check is only useful if the variable (`this`) may not be defined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#typeof_operator_and_undefined), but `this` is always defined in this context. So the `typeof` check is redundant.

Thus, the current check is fine.

Reply:
With loose equality check `==` I guess you're right 👍 
I always prefer strict equality check `===` and in that case this won't work.

And I also noticed we have in the code `typeof window !== 'undefined'` not `window != null` right?

Reply:
Since `window` variable may not exist, e.g., in SSR, `typeof window !== 'undefined'` comes in.